src/util/sfdcUrl.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { URL } from 'url';
+import { Env, Duration } from '@salesforce/kit';
+import { MyDomainResolver } from '../status/myDomainResolver';
+import cache from './cache';
+
+export class SfdcUrl extends URL {
+  /**
+   * Salesforce URLs.
+   */
+  public static SANDBOX = 'https://test.salesforce.com';
+  public static PRODUCTION = 'https://login.salesforce.com';
+
+  public constructor(input: string | URL, base?: string | URL) {
+    super(input.toString(), base);
+    if (this.protocol !== 'https:' && !cache.has(this.origin)) {
+      cache.set(this.origin, {
+        timestamp: new Date().getTime(),
```

Comment:
I know we discussed timestamping these, but I think that was before we used the singleton-cache.  We're not doing anything with the timestamps, so that key value could be empty?

Probably doesn't matter.